// gatsby/dist/state-machines/data-layer/actions.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

exports.__esModule = true;
exports.dataLayerActions = exports.assignGatsbyNodeGraphQL = exports.assignChangedPages = void 0;

var _xstate = require("xstate");

var _createGraphqlRunner = require("../../bootstrap/create-graphql-runner");

var _reporter = _interopRequireDefault(require("gatsby-cli/lib/reporter"));

var _assertStore = require("../../utils/assert-store");

const concatUnique = (array1 = [], array2 = []) => Array.from(new Set(array1.concat(array2)));

const assignChangedPages = (0, _xstate.assign)((context, event) => {
  return {
    pagesToBuild: concatUnique(context.pagesToBuild, event.data.changedPages),
    pagesToDelete: concatUnique(context.pagesToDelete, event.data.deletedPages)
  };
});
exports.assignChangedPages = assignChangedPages;
const assignGatsbyNodeGraphQL = (0, _xstate.assign)({
  gatsbyNodeGraphQLFunction: ({
    store
  }) => {
    (0, _assertStore.assertStore)(store);
    return (0, _createGraphqlRunner.createGraphQLRunner)(store, _reporter.default);
  }
});
exports.assignGatsbyNodeGraphQL = assignGatsbyNodeGraphQL;
const dataLayerActions = {
  assignChangedPages,
  assignGatsbyNodeGraphQL
};
exports.dataLayerActions = dataLayerActions;
