// gatsby/dist/services/index.js
"use strict";

exports.__esModule = true;
var _exportNames = {
  buildServices: true,
  startWebpackServer: true,
  rebuildSchemaWithSitePage: true,
  extractQueries: true,
  writeOutRedirects: true,
  postBootstrap: true,
  buildSchema: true,
  createPages: true,
  createPagesStatefully: true,
  customizeSchema: true,
  initialize: true,
  sourceNodes: true,
  writeOutRequires: true,
  calculateDirtyQueries: true,
  runStaticQueries: true,
  runPageQueries: true,
  waitUntilAllJobsComplete: true
};
exports.buildServices = void 0;

var _startWebpackServer = require("./start-webpack-server");

exports.startWebpackServer = _startWebpackServer.startWebpackServer;

var _rebuildSchemaWithSitePages = require("./rebuild-schema-with-site-pages");

exports.rebuildSchemaWithSitePage = _rebuildSchemaWithSitePages.rebuildSchemaWithSitePage;

var _extractQueries = require("./extract-queries");

exports.extractQueries = _extractQueries.extractQueries;

var _writeOutRedirects = require("./write-out-redirects");

exports.writeOutRedirects = _writeOutRedirects.writeOutRedirects;

var _postBootstrap = require("./post-bootstrap");

exports.postBootstrap = _postBootstrap.postBootstrap;

var _buildSchema = require("./build-schema");

exports.buildSchema = _buildSchema.buildSchema;

var _createPages = require("./create-pages");

exports.createPages = _createPages.createPages;

var _createPagesStatefully = require("./create-pages-statefully");

exports.createPagesStatefully = _createPagesStatefully.createPagesStatefully;

var _customizeSchema = require("./customize-schema");

exports.customizeSchema = _customizeSchema.customizeSchema;

var _initialize = require("./initialize");

exports.initialize = _initialize.initialize;

var _sourceNodes = require("./source-nodes");

exports.sourceNodes = _sourceNodes.sourceNodes;

var _writeOutRequires = require("./write-out-requires");

exports.writeOutRequires = _writeOutRequires.writeOutRequires;

var _calculateDirtyQueries = require("./calculate-dirty-queries");

exports.calculateDirtyQueries = _calculateDirtyQueries.calculateDirtyQueries;

var _runStaticQueries = require("./run-static-queries");

exports.runStaticQueries = _runStaticQueries.runStaticQueries;

var _runPageQueries = require("./run-page-queries");

exports.runPageQueries = _runPageQueries.runPageQueries;

var _waitUntilJobsComplete = require("../utils/wait-until-jobs-complete");

exports.waitUntilAllJobsComplete = _waitUntilJobsComplete.waitUntilAllJobsComplete;

var _types = require("./types");

Object.keys(_types).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  exports[key] = _types[key];
});
const buildServices = {
  customizeSchema: _customizeSchema.customizeSchema,
  sourceNodes: _sourceNodes.sourceNodes,
  createPages: _createPages.createPages,
  buildSchema: _buildSchema.buildSchema,
  createPagesStatefully: _createPagesStatefully.createPagesStatefully,
  extractQueries: _extractQueries.extractQueries,
  writeOutRequires: _writeOutRequires.writeOutRequires,
  calculateDirtyQueries: _calculateDirtyQueries.calculateDirtyQueries,
  runStaticQueries: _runStaticQueries.runStaticQueries,
  runPageQueries: _runPageQueries.runPageQueries,
  initialize: _initialize.initialize,
  waitUntilAllJobsComplete: _waitUntilJobsComplete.waitUntilAllJobsComplete,
  postBootstrap: _postBootstrap.postBootstrap,
  writeOutRedirects: _writeOutRedirects.writeOutRedirects,
  startWebpackServer: _startWebpackServer.startWebpackServer,
  rebuildSchemaWithSitePage: _rebuildSchemaWithSitePages.rebuildSchemaWithSitePage
};
exports.buildServices = buildServices;
